lib/chef/licensing.rb

```diff
@@ -0,0 +1,94 @@
+require "chef-licensing"
+require_relative "log"
+require_relative "licensing_config"
+
+class Chef
+  class Licensing
+    class << self
+      def fetch_and_persist
+        Chef::Log.info "Fetching and persisting license..."
+        license_keys = ChefLicensing.fetch_and_persist
+      rescue ChefLicensing::LicenseKeyFetcher::LicenseKeyNotFetchedError
+        Chef::Log.error "Chef Infra cannot execute without valid licenses." # TODO: Replace Infra with the product name dynamically
+        Chef::Application.exit! "License not set", 174 # 174 is the exit code for LICENSE_NOT_SET defined in lib/chef/application/exit_code.rb
```

Comment:
Shouldn't this use the name, not the number? Chef::Application::ExitCode::VALID_RFC_062_EXIT_CODES::LICENSE_NOT_SET

Reply:
This is not needed in this PR and removed it.